crypto/asn1/asn1_lib.c

```diff
@@ -275,18 +275,24 @@ ASN1_STRING *ASN1_STRING_dup(const ASN1_STRING *str)
     return ret;
 }
 
-int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len)
+int ASN1_STRING_set(ASN1_STRING *str, const void *_data, int len_in)
 {
     unsigned char *c;
     const char *data = _data;
+    size_t len;
 
-    if (len < 0) {
+    if (len_in < 0) {
         if (data == NULL)
             return 0;
-        else
-            len = strlen(data);
+        len = strlen(data);
+    } else {
+        len = (size_t)len_in;
+    }
+    if (len > INT_MAX - 1) {
```

Comment:
`- 1` is to leave space for an ending NUL byte, right?  You might want to add a quick comment to that effect

Reply:
Actually it would not necessarily have to be there because the length does not cover the extra NUL byte however I think it is reasonable to have it there "just-in-case" someone uses the length value with int variable and tries to increment it.

Reply:
I figured out that the `-1` wasn't vital but allowing the user to `strlen()+1` and store into an integer seemed worthwhile as a surprise avoider.

I've added a comment explaining the situation.